src/app/tree-grid/tree-grid-validator-service-extended/tree-grid-validator-service-extended.component.html

```diff
@@ -0,0 +1,36 @@
+<div class="grid__wrapper">
+    <igx-tree-grid igxPreventDocumentScroll #treeGrid [batchEditing]="true" [data]="data" primaryKey="ID" [autoGenerate]="false"
+        foreignKey="ParentID" [width]="'100%'" [height]="'500px'" [rowEditable]="true"
+        (formGroupCreated)="formCreateHandler($event)">
+        <igx-column field="ID" header="ID" dataType="number" [editable]="false"></igx-column>
+        <igx-column field="Age" header="Age" dataType="number" minAge="18" [editable]="true" required max="99">
+        </igx-column>
+        <igx-column field="Name" header="Full Name" dataType="string" [editable]="true" required>
+        </igx-column>
+        <igx-column field="Phone" header="Phone" dataType="string" [editable]="true" required phoneFormat="^[^a-zA-Z]*$">
+            <ng-template igxCellValidationError let-cell='cell'>
```

Comment:
This won't show an error message for `required` (for  this and HireDate column as well). Either use the default error template or add a custom one for required below.